app/src/main/java/com/codepath/apps/critterfinder/adapters/PetFavoritesAdapter.java

```diff
@@ -53,15 +57,29 @@ public ViewHolder(final View itemView) {
             tvPetFavSize = (TextView) itemView.findViewById(R.id.petFavSize);
             tvPetFavAge = (TextView) itemView.findViewById(R.id.petFavAge);
             tvPetFavBreeds = (TextView) itemView.findViewById(R.id.petFavBreeds);
+            ibRemovePetPav = (ImageButton) itemView.findViewById(R.id.ibRemovePetFav);
+            ibRemovePetPav.setOnClickListener(this);
             itemView.setOnClickListener(this);
+
         }
 
         @Override
         public void onClick(View v) {
-            if (mOnItemClickListener != null) {
-                mOnItemClickListener.onItemClick(v, ivPetFavImage, getAdapterPosition());
+            if (v instanceof ImageButton){
+                mOnItemClickListener.onButtonClicked((ImageButton)v,this,getAdapterPosition());
+            }else {
+
+                if (mOnItemClickListener != null) {
+                    mOnItemClickListener.onItemClick(v, ivPetFavImage, getAdapterPosition());
+                }
```

Comment:
Good stuff! Just a tip - considering just calling 

```
mOnItemClickListener.onItemClick(v, ivPetFavImage, getAdapterPosition());
```
instead of having a separate onButtonClicked method.

The activity's onItemClick can check the id of the view and dispatch to the correct action. That scales up better if you've got a lot of touchable targets on your view.